src/coord/src/coord.rs

```diff
@@ -320,20 +291,25 @@ pub struct Coordinator {
     /// active connections.
     active_conns: HashMap<u32, ConnMeta>,
 
-    /// Holds pending compaction messages to be sent to the dataflow workers. When
-    /// `since_handles` are advanced or `txn_reads` are dropped, this can advance.
-    index_since_updates: Rc<RefCell<HashMap<GlobalId, Antichain<Timestamp>>>>,
-    /// Holds pending compaction messages to be sent to the dataflow workers. When
-    /// `since_handles` are advanced or `txn_reads` are dropped, this can advance.
-    source_since_updates: Rc<RefCell<HashMap<GlobalId, Antichain<Timestamp>>>>,
-    /// Holds handles to ids that are advanced by update_upper.
-    since_handles: HashMap<GlobalId, AntichainToken<Timestamp>>,
-    /// Tracks active read transactions so that we don't compact any indexes beyond
-    /// an in-progress transaction.
-    // TODO(mjibson): Should this live on a Session?
-    txn_reads: HashMap<u32, TxnReads>,
-    /// Tracks write frontiers for active exactly-once sinks.
-    sink_writes: HashMap<GlobalId, SinkWrites<Timestamp>>,
+    /// For each identifier, its read policy and any transaction holds on time.
+    ///
+    /// The antichain contains the accumulation of all expressed pins, and prevents
+    /// compaction past its lower bound, to ensure they all remain valid. The antichain
+    /// is emptied only when all transactions have dropped, at which point the read
+    /// policy for the identifier is returned to its current compaction policy.
+    read_capability_needs: HashMap<GlobalId, ReadCapabilityState<mz_repr::Timestamp>>,
```

Comment:
Comment and naming could use a refresh, now that there are not competing concepts in `Coordinator`.